addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java

```diff
@@ -450,13 +482,30 @@ void shouldRaiseAlertBelowHighThresholdAndOutOfScope(AlertThreshold threshold)
         assertEquals(1, alertsRaised.size());
     }
 
+    private static Stream<Arguments> createMethodAndScopeBoolean() {
+        List<Arguments> args = new ArrayList<>();
+        args.addAll(getListForMethod(POSTS));
+        args.addAll(getListForMethod(GETS));
+        return args.stream();
+    }
+
+    private static List<Arguments> getListForMethod(List<String> methods) {
+        List<Arguments> args = new ArrayList<>();
+        for (String method : methods) {
+            for (boolean bool : BOOLS) {
+                args.add(Arguments.of(method, bool));
+            }
+        }
+        return args;
+    }
+
     @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    void shouldRaiseAlertOnGetAtLowThresholdRegardlessOfScope(boolean isInScope)
+    @MethodSource("createMethodAndScopeBoolean")
+    void shouldRaiseAlertOnGetAtLowThresholdRegardlessOfScope(String method, boolean isInScope)
```

Comment:
It's preferable to have specific tests testing specific conditions rather than trying to do all at once. Some of the test names no longer match what they are testing also would be better to have a specific method case in other tests rather than randomly changing the case.